addon/-private/system/store.js

```diff
@@ -1812,15 +1828,15 @@ Store = Service.extend({
     @private
   */
   flushPendingSave() {
-    var pending = this._pendingSave.slice();
+    let pending = this._pendingSave.slice();
     this._pendingSave = [];
 
     pending.forEach((pendingItem) => {
-      var snapshot = pendingItem.snapshot;
-      var resolver = pendingItem.resolver;
-      var record = snapshot._internalModel;
-      var adapter = this.adapterFor(record.type.modelName);
-      var operation;
+      let snapshot = pendingItem.snapshot;
+      let resolver = pendingItem.resolver;
+      let record = snapshot._internalModel;
+      let adapter = this.adapterFor(record.modelClass.modelName);
```

Comment:
Actually disregard, this is an internalModel and forward thinking me has already made that work.